File: tests/fixtures/polyglot/sample.cpp
// Polyglot fixture — use-after-move.
// `s` is moved into `sink` on line 11; read again on line 12 is undefined.
// Parses clean; clang-tidy bugprone-use-after-move would flag line 12.

#include <string>
#include <iostream>
#include <utility>

void sink(std::string) {}

int main() {
    std::string s = "hello";
    sink(std::move(s));
    std::cout << s.size() << "\n";
    return 0;
}

File: tests/fixtures/polyglot/sample.ts
// Polyglot fixture — null-deref style.
// `user` is typed `User | null`; `user.name` on line 12 is dereffed without a guard.
// Parses clean; semgrep's ts.null-deref ruleset would flag line 12 if present.

interface User {
    id: number;
    name: string;
}

function getUser(id: number): User | null {
    return null;
}

function greet(id: number): string {
    const user = getUser(id);
    return "hi " + user.name;
}

greet(7);
